File: practica3/python/practica3.py
class Pila:
    def __init__(self, n):
        self.n = n 
        self.arreglo = [0] * n 
        self.top = -1 

    def push(self, e):
        if self.isFull():
            print("La pila está llena, no se puede agregar más elementos.")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            print("La pila está vacía, no se puede sacar elementos.")
            return None
        else:
            elemento = self.arreglo[self.top]
            self.top -= 1
            return elemento

    def peek(self):
        if self.isEmpty():
            print("La pila está vacía.")
            return None
        return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

# Prueba de la Pila
pila = Pila(5)
pila.push(10)
pila.push(20)
print("Elemento en la cima:", pila.peek())   
print("Elemento sacado:", pila.pop())  
print("La pila está vacía?", pila.isEmpty()) 

File: practica3/practica3.java
package practica4;

public class practica3 {
    private long[] arreglo;
    private int top;
    private int n;

    public practica3(int n) {
        this.n = n;
        this.arreglo = new long[n];  // Crear un arreglo de tamaño n
        this.top = -1;  // Indica la posición del último elemento
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("La pila está llena, no se puede agregar más elementos.");
        } else {
            top++;
            arreglo[top] = e;
        }
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("La pila está vacía, no se puede sacar elementos.");
            return -1;
        } else {
            return arreglo[top--];
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("La pila está vacía.");
            return -1;
        }
        return arreglo[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        practica3 pila = new practica3(5);
        pila.push(10);
        pila.push(20);
        System.out.println("Elemento en la cima: " + pila.peek());  // Debería mostrar 20
        System.out.println("Elemento sacado: " + pila.pop());  // Debería mostrar 20
        System.out.println("¿La pila está vacía? " + pila.isEmpty());  // Debería mostrar false
    }
}
